packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -284,6 +299,72 @@ class _ScrollableScope extends InheritedWidget {
   }
 }
 
+/// A typedef for a function that can calculate the offset for a type of scroll
+/// increment given a [ScrollIncrementDetails].
+typedef ScrollIncrementCalculator = double Function(ScrollIncrementDetails details);
+
+/// Describes the type of scroll increment that will be performed by
+/// [ScrollAction] on a [Scrollable].
+///
+/// This is used to configure a [ScrollIncrementDetails] object to pass to a
+/// [ScrollIncrementCalculator] function on a [Scrollable].
+enum ScrollIncrementType {
+  /// Indicates that the [ScrollIncrementCalculator] should return the scroll
+  /// distance it should move when the user requests to scroll by a "line".
+  ///
+  /// The distance a "line" scrolls is really refers to what should happen when
+  /// the key binding for "scroll down/up by a line" is triggered. It's up to
+  /// the [ScrollIncrementCalculator] function to decide what that means for a
+  /// particular scrollable.
+  line,
+
+  /// Indicates that the [ScrollIncrementCalculator] should return the scroll
+  /// distance it should move when the user requests to scroll by a "page".
+  ///
+  /// The distance a "page" scrolls is really refers to what should happen when
+  /// the key binding for "scroll down/up by a page" is triggered. It's up to
+  /// the [ScrollIncrementCalculator] function to decide what that means for a
+  /// particular scrollable.
+  page,
+}
+
+/// A details object that describes the type of scroll increment being requested
+/// of a [ScrollIncrementCalculator] function.
+class ScrollIncrementDetails {
```

Comment:
It seems strange to me that this isn't implementing or extending `ScrollMetrics`.

Reply:
Aah.  I didn't know about `ScrollMetrics`, that's why.  Now that I'm looking at how that is used, would it make sense to derive this from `ScrollMetrics`, and then make a specialized `ScrollActivity` that does this kind of scrolling instead of using `moveTo` directly on the scroll position in my `ScrollAction` class? Or is that strictly for touch interactions?

Reply:
I think that makes sense - I can't think of any reason why `ScrollActivity would be exclusively for touch.

Reply:
Actually, it appears that `ScrollActivity` is something that receives updates about the current scroll position, rather than driving the scroll position, so it's not clear to me that it is the right path to take.  I addressed your original concern by just putting a `ScrollMetrics` into the details.  A `ScrollPosition` implements `ScrollMetrics`, so it's easy to pass one in.